tests/data_analyzer/test_beta_2_mode.py

```diff
@@ -340,3 +342,122 @@ def test_DataAnalyzerProcess__formats_and_passes_incoming_stim_packet_through_to
 
     assert outgoing_msg["data_type"] == "stimulation"
     assert outgoing_msg["data_json"] == json.dumps(expected_stim_data)
+
+
+def test_DataAnalyzerProcess__correctly_handles_command_from_main_to_start_mag_analysis(
+    four_board_analyzer_process_beta_2_mode, mocker
+):
+
+    da_process = four_board_analyzer_process_beta_2_mode["da_process"]
+    from_main_queue = four_board_analyzer_process_beta_2_mode["from_main_queue"]
+
+    spied_start_analysis = mocker.spy(da_process, "_start_mag_finding_analysis")
+    mocker.patch.object(data_analyzer, "_mag_finding_analysis_thread", autospec=True)
+
+    test_mag_analysis_command = copy.deepcopy(TEST_START_MAG_ANALYSIS_COMMAND)
+
+    put_object_into_queue_and_raise_error_if_eventually_still_empty(
+        test_mag_analysis_command, from_main_queue
+    )
+    invoke_process_run_and_check_errors(da_process, 1)
+    spied_start_analysis.assert_called_once_with(TEST_START_MAG_ANALYSIS_COMMAND["content"])
+
+    drain_queue(from_main_queue)
+
+
+def test_DataAnalyzerProcess__correctly_handles_command_from_main_to_prevent_start_if_beta_1(
+    runnable_four_board_analyzer_process, mocker
+):
+
+    da_process, _, from_main_queue, _, _, _ = runnable_four_board_analyzer_process
+
+    spied_start_analysis = mocker.spy(da_process, "_start_mag_finding_analysis")
+    mocker.patch.object(data_analyzer, "_mag_finding_analysis_thread", autospec=True)
+
+    test_mag_analysis_command = copy.deepcopy(TEST_START_MAG_ANALYSIS_COMMAND)
+
+    put_object_into_queue_and_raise_error_if_eventually_still_empty(
+        test_mag_analysis_command, from_main_queue
+    )
+    invoke_process_run_and_check_errors(da_process, 1)
+    spied_start_analysis.assert_not_called()
+
+    drain_queue(from_main_queue)
+
+
+def test_DataAnalyzerProcess__correctly_handles_thread_status_with_failures(
+    four_board_analyzer_process_beta_2_mode, mocker
+):
+
+    da_process = four_board_analyzer_process_beta_2_mode["da_process"]
+    from_main_queue = four_board_analyzer_process_beta_2_mode["from_main_queue"]
+    to_main_queue = four_board_analyzer_process_beta_2_mode["to_main_queue"]
+    mag_analysis_output_dir = four_board_analyzer_process_beta_2_mode["mag_analysis_output_dir"]
+
+    spied_check_analysis = mocker.spy(da_process, "_check_mag_analysis_statuses")
+
+    mocked_pr = mocker.patch.object(PlateRecording, "from_directory", autospec=True)
+    mocked_pr.return_value = iter([PlateRecording])
+
+    mocker.patch.object(PlateRecording, "write_time_force_csv", side_effect=Exception("exception raised"))
+
+    test_mag_analysis_command = copy.deepcopy(TEST_START_MAG_ANALYSIS_COMMAND)
+    put_object_into_queue_and_raise_error_if_eventually_still_empty(
+        test_mag_analysis_command, from_main_queue
+    )
+    invoke_process_run_and_check_errors(da_process, 7)
+    assert len(spied_check_analysis.call_args_list) == 7
+
+    msg = to_main_queue.get()
```

Comment:
use `get_nowait()` or `get(timeout=QUEUE_CHECK_TIMEOUT_SECONDS)` since `get()` will block if there are no items in the queue